src/sync.cpp

```diff
@@ -94,35 +94,33 @@ LockData& GetLockData() {
 
 static void potential_deadlock_detected(const LockPair& mismatch, const LockStack& s1, const LockStack& s2)
 {
-    std::string strOutput = "";
-    strOutput += "POTENTIAL DEADLOCK DETECTED\n";
-    strOutput += "Previous lock order was:\n";
+    LogPrintf("POTENTIAL DEADLOCK DETECTED\n");
+    LogPrintf("Previous lock order was:\n");
     for (const LockStackItem& i : s1) {
+        std::string prefix{};
         if (i.first == mismatch.first) {
-            strOutput += " (1)";
+            prefix = " (1)";
         }
         if (i.first == mismatch.second) {
-            strOutput += " (2)";
+            prefix = " (2)";
         }
-        strOutput += strprintf(" %s\n", i.second.ToString());
+        LogPrintf("%s %s\n", prefix, i.second.ToString());
     }
 
     std::string mutex_a, mutex_b;
-    strOutput += "Current lock order is:\n";
+    LogPrintf("Current lock order is:\n");
     for (const LockStackItem& i : s2) {
+        std::string prefix{};
         if (i.first == mismatch.first) {
-            strOutput += " (1)";
+            prefix = " (1)";
             mutex_a = i.second.Name();
         }
         if (i.first == mismatch.second) {
-            strOutput += " (2)";
+            prefix = " (2)";
             mutex_b = i.second.Name();
         }
-        strOutput += strprintf(" %s\n", i.second.ToString());
+        LogPrintf("%s %s\n", prefix, i.second.ToString());
     }
-
-    LogPrintf("%s\n", strOutput);
-
     if (g_debug_lockorder_abort) {
```

Comment:
23104: that's a regression imo

Reply:
I also prefer the code as it exists in `develop` but since the downsides of the upstream approach have been mostly mitigated by the backport, upstream realignment now weighs greater imo.

Reply:
upstream fix is minimal - it still uses tons of `LogPrintf`s so the output is going to be scattered all over the place, I'd rather avoid that